lib/swoosh/mailer.ex

```diff
@@ -56,19 +56,28 @@ defmodule Swoosh.Mailer do
 
       def __adapter__, do: @adapter
 
-      def deliver(%Swoosh.Email{from: nil}), do: raise ArgumentError, "expected \"from\" to be set"
-      def deliver(%Swoosh.Email{from: {_name, nil}}), do: raise ArgumentError, "expected \"from\" address to be set"
-      def deliver(%Swoosh.Email{to: nil}), do: raise ArgumentError, "expected \"to\" to be set"
-      def deliver(%Swoosh.Email{to: []}), do: raise ArgumentError, "expected \"to\" not to be empty"
-      def deliver(%Swoosh.Email{subject: nil}), do: raise ArgumentError, "expected \"subject\" to be set"
-      def deliver(%Swoosh.Email{html_body: nil, text_body: nil}) do
+      def deliver(email, config \\ [])
+      def deliver(%Swoosh.Email{from: nil}, _config), do:
+        raise ArgumentError, "expected \"from\" to be set"
+      def deliver(%Swoosh.Email{from: {_name, nil}}, _config), do:
+        raise ArgumentError, "expected \"from\" address to be set"
+      def deliver(%Swoosh.Email{to: nil}, _config), do:
+        raise ArgumentError, "expected \"to\" to be set"
+      def deliver(%Swoosh.Email{to: []}, _config), do:
```

Comment:
It's really unreadable, happy to explore other options but this is by far the most readable